src/stan-users-guide/odes.Rmd

```diff
@@ -257,124 +166,94 @@ Note that when the system is linear, it is a good idea to try
 using the matrix exponential function, as the method can, depending
 on the system at hand, yield more efficient results.
 
-
 This program illustrates the way in which the ODE solver is called in
 a Stan program,
 
 ```
-y_hat = integrate_ode_rk45(sho, y0, t0, ts, theta, x_r, x_i);
+vector[2] y_hat[T] = ode_rk45(sho, y0, t0, ts, theta);
 ```
 
-This assigns the solutions to the system defined by function
-`sho`, given initial state `y0`, initial time `t0`,
-requested solution times `ts`, parameters `theta`, real data
-`x`, and integer data `x_int`.  The call explicitly
+this returns the solution of the ODE initial value problem defined
+by system function `sho`, initial state `y0`, initial time `t0`, and
+parameter `theta` at the times `ts`. The call explicitly
 specifies the Runge-Kutta solver (for non-stiff systems).
 
 Here, the ODE solver is called in the generated quantities block to
-provide a $10 \times 2$ array of solutions `y_hat` to
-which measurement error is added using the normal pseudo-random number
-generating function `normal_rng`.  The number of rows in the
-solution array is the same as the size of `ts`, the requested
-solution times.
-
-### Data versus Parameters {-}
```

Comment:
I removed this because the new interface doesn't have this limitation (it can take any combination of data/param arguments).

In the new interface you might accidentally over-promote a data thing by defining it in transformed parameters, but whether it is defined in transformed parameters or transformed data the solver will still work.